### ISSUE TITLE
SA allows members or auditors to edit slice membership

### ISSUE THREAD

**Issue by ahelsing** [GENI-NSF/geni-portal#533]
I made myself an auditor of a slice.
The slice page properly greys out the Edit Slice Membership button.

Then I manually went to the edit slice membership page. It loaded fine.
I made myself an Admin.
It worked. It should not have worked.

 _Imported from trac ticket [#533](http://trac.gpolab.bbn.com/proto-ch/ticket/533),  created by ahelsing on 05-15-2013 at 10:32, last modified: 09-06-2013 at 14:04_

**Comment by ahelsing** [comment]
Note this was tested after manually re-applying the latest DB updates

_Trac comment by ahelsing on 05-15-2013 at 10:32_

**Comment by seledwards** [comment]
Ok.  I confirmed this bug on my dev machine.

_Trac comment by sedwards (github user: seledwards) on 05-15-2013 at 10:34_

**Comment by ahelsing** [comment]
More details:
I see this when the Project Lead changes themselves from being a slice admin to an auditor, and then changes themselves to a slice admin.
I do _not_ see this when the Slice Owner/Lead (not a project lead) changes themselves to an auditor, and then tries to change themselves to a slice admin.


_Trac comment by ahelsing on 05-15-2013 at 11:26_

**Comment by ahelsing** [comment]
This may be fixed.

_Trac comment by ahelsing on 08-01-2013 at 14:39_

**Comment by ahelsing** [comment]
In a project where I was not lead, I created a slice. Then I edited the slice membership, making myself an auditor.
Then I manually loaded the edit slice membership page, and was able to make myself the slice owner.
This bug still exists.

 1. edit-slice-membership should check the permissions of the caller
 2. The SA should check the permissions of the caller making the change

_Trac comment by ahelsing on 08-13-2013 at 10:59_
